### PR TITLE
Misc warnings

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: src/iso19111/c_api.cpp [hunk 1]
```diff
@@ -289,7 +289,6 @@ PJ_OBJ_LIST::~PJ_OBJ_LIST() = default;
  * @param ctx Ignored
  * @param autoclose Ignored
  * @since 6.2
- * @deprecated Since 8.1
```
Comment (rouault) [inline src/iso19111/c_api.cpp:292]: Please keep it as a regular Doxygen warning then, without the '@'

File: src/list.cpp [hunk 2]
```diff
@@ -1,10 +1,10 @@
 /* Projection System: default list of projections
-** Use local definition of PJ_LIST_H for subset.
+** Use local definition of PJ_LIST_H (should this be USE_PJ_LIST_H ??) for subset.
 */
 
 #include "proj.h"
 
-#define USE_PJ_LIST_H 1
+#define USE_PJ_LIST_H 1 // -Wunused-macro here. Indeed "USE_PJ_LIST_H" appears nowhere else in code
```
Comment (rouault) [inline src/list.cpp]: could be removed
```suggestion
```

File: src/list.cpp [hunk 3]
```diff
@@ -1,10 +1,10 @@
 /* Projection System: default list of projections
-** Use local definition of PJ_LIST_H for subset.
+** Use local definition of PJ_LIST_H (should this be USE_PJ_LIST_H ??) for subset.
```
Comment (rouault) [inline src/list.cpp]:
```suggestion
```

File: src/iso19111/c_api.cpp (hunk 1 above)
Comment (seanm) [reply src/iso19111/c_api.cpp:292]: Alternatively, does PROJ have a mechanism to mark things deprecated?  For example, is the [C++14 `[[deprecated]]`](https://en.cppreference.com/w/cpp/language/attributes/deprecated) too new?

Comment (rouault) [reply src/iso19111/c_api.cpp:292]: > For example, is the [C++14 [[deprecated]]](https://en.cppreference.com/w/cpp/language/attributes/deprecated) too new?

yes, we stick to C++11 for now

File: src/list.cpp (hunk 2 above)
Comment (seanm) [reply src/list.cpp]: Removed.

File: src/iso19111/c_api.cpp (hunk 1 above)
Comment (seanm) [reply src/iso19111/c_api.cpp:292]: OK, I just removed the `@` character then.

**Comment by seanm** [comment]
Oh, didn't expect you to merge yet, one of the commits was still marked WIP...

**Comment by rouault** [comment]
> Oh, didn't expect you to merge yet, one of the commits was still marked WIP...

ah, didn't notice the WIP (best to use to use the github feature to tag the pull request as Draft then). Our CI has Windows workers. What is unknown if that works with UWP for which we don't have CI. @gorlak does https://github.com/OSGeo/PROJ/pull/3424/commits/fecca6f2a1619347a9026cdb21a3576a1e550ef6 look good to you ?

**Comment by gorlak** [comment]
That change makes sense to me 👍

**Pull request merged**